### PR TITLE
[YUNIKORN-2513] Fix event system to use event.requestCapacity

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
`request.eventCapacity` is completely ignored now, even though it is in the design doc. The PR changes code to read & use this property.  

### What type of PR is it?
* [x] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2513

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/events/event_store.go [hunk 1]
```diff
@@ -21,31 +21,31 @@ package events
...
 //   - there is a cap for the number of events stored
 //   - the CollectEvents() function clears the currently stored events in the EventStore
 //
 // Assuming the rate of events generated by the scheduler component in a given time period
 // is high, calling CollectEvents() periodically should be fine.
 type EventStore struct {
-	events []*si.EventRecord
-	idx    int // points where to store the next event
+	events   []*si.EventRecord
+	idx      int // points where to store the next event
```
Comment (wilfred-s) [inline pkg/events/event_store.go:39]: this means we cannot use the fulll size as that is an unit64.Should we not have the same types?

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: pkg/events/event_store.go [hunk 2]
```diff
@@ -68,8 +68,12 @@ func (es *EventStore) CollectEvents() []*si.EventRecord {
 	defer es.Unlock()
 
 	messages := es.events[:es.idx]
-	es.events = make([]*si.EventRecord, defaultEventStoreSize)
+	if es.size != es.lastSize {
+		log.Log(log.Events).Info("Resizing event store", zap.Uint64("last", es.lastSize), zap.Uint64("new", es.size))
+	}
+	es.events = make([]*si.EventRecord, es.size)
```
Comment (chia7712) [inline pkg/events/event_store.go:74]: Do we need to reallocate array if the `idx` can be used to guard against incorrect reading.

**Review by chia7712 — COMMENTED** [review]
@pbacsko nice patch! a couple of trivial comments left. PTAL

File: pkg/events/event_system.go [hunk 3]
```diff
@@ -233,22 +235,36 @@ func (ec *EventSystemImpl) AddEvent(event *si.EventRecord) {
...
+func isTrackingEnabled() bool {
 	return common.GetConfigurationBool(configs.GetConfigMap(), configs.CMEventTrackingEnabled, configs.DefaultEventTrackingEnabled)
 }
 
-func (ec *EventSystemImpl) readRequestCapacity() int {
-	return common.GetConfigurationInt(configs.GetConfigMap(), configs.CMEventRequestCapacity, configs.DefaultEventRequestCapacity)
+func getRequestCapacity() uint64 {
+	capacity := common.GetConfigurationUint(configs.GetConfigMap(), configs.CMEventRequestCapacity, configs.DefaultEventRequestCapacity)
+	if capacity == 0 {
+		log.Log(log.Events).Warn("Request capacity is set to 0, using default",
```
[the remark this answers is not in the document]
Comment (pbacsko) [reply pkg/events/event_system.go:245]: it's uint64, cannot be lower than 0

File: pkg/events/event_system.go [hunk 4]
```diff
@@ -272,18 +288,22 @@ func (ec *EventSystemImpl) CloseAllStreams() {
 }
 
 func (ec *EventSystemImpl) reloadConfig() {
-	ec.updateRequestCapacity()
+	requestCapacity, ringBufferCapacity := ec.updateCapacity()
```
[the remark this answers is not in the document]
Comment (pbacsko) [reply pkg/events/event_system.go]: update is now done inside `reloadConfig()`

File: pkg/events/event_store.go (hunk 2 above)
Comment (pbacsko) [reply pkg/events/event_store.go]: correct, it's not needed, new code does not reallocate slice storage every time

**Comment by chenyulin0719** [comment]
@pbacsko The title for this Jira and PR is incorrect. 
"request.eventCapacity" -> "event.requestCapacity"

**Review by chenyulin0719 — APPROVED** [review]
+1 LGTM

**Pull request closed without merging**